### ISSUE TITLE
Mixing RTL and LTR characters messes up the display order of text

### ISSUE THREAD

**Issue by saad440** [vector-im/riot-web#2922]
Riot currently doesn't have good support for RTL languages. Complete lines of RTL characters are somewhat manageable but when there's a sequence of LTR characters in the middle, it messes up the sequence to display the words in.
Perhaps Riot should detect whether a line starts with RTL characters and then treat it all as RTL.

**Comment by richvdh** [comment]
Related to #1712. Needs more `<span dir="auto">`.

**Comment by saad440** [comment]
Take this text, for example:
یہ ایک مثال ہے۔ درمیان میں left-to-right متن آنے پر متن کی ترتیب بدل جاتی ہے۔
(Not displaying correctly even on github)
The way it should show up is
![selection_054](https://cloud.githubusercontent.com/assets/10145377/21862562/9302702e-d85b-11e6-952a-135aa5378e05.png)
But it shows up in Riot as
![selection_053](https://cloud.githubusercontent.com/assets/10145377/21862574/9b0a9da0-d85b-11e6-809f-f4a01f39c769.png)

**Comment by Hitaro** [comment]
The web & Desktop version of Riot does not support RTL languages such as Arabic, Persian, Hebrew, etc. While the Android app can detect the language automatically and things are shown properly.

For example, if you put an exclamation at the end of مرحبا, it will be shown at the first of the word. **Look, we read/write from right side to left, so the end of a sentence is at the left side**.

`مرحبا!`

**Comment by lukebarnard1** [comment]
This looks like it's been fixed, Riot now displays
![2017-08-09-132954_493x29_scrot](https://user-images.githubusercontent.com/6750344/29121551-dfd83b04-7d06-11e7-9b72-33256f9a0088.png)

**Comment by saad440** [comment]
Yes, it is working in Firefox. Let me check in Chrome and the electron app.

**Comment by saad440** [comment]
Also fixed in the mobile app.

**Comment by richvdh** [comment]
fixed in https://github.com/matrix-org/matrix-react-sdk/pull/955 I think \o/

**Comment by saad440** [comment]
Fixed in Chrome. So I guess it will be fixed in the electron app too. Both are webkit, right?

**Comment by richvdh** [comment]
@saad440: yes.

**Comment by richvdh** [comment]
Shame we don't right-justify the text: opened #4771 to track it

**Comment by saad440** [comment]
Fixed in the electron app too.
